src/gui/mzroll/autoupdater.cpp

```diff
@@ -0,0 +1,124 @@
+#define _STR(X) #X
+#define STR(X) _STR(X)
+
+#include <QProcess>
+#include <QDebug>
+#include <QApplication>
+
+#include <pugixml.hpp>
+
+#include <common/downloadmanager.h>
+#include "autoupdater.h"
+#include "base64.h"
+#include "projectversioning.h"
+
+void AutoUpdater::startMaintenanceTool()
+{
+    // Windows: <install_dir>/El-MAVEN/bin/
+    // MacOS:   <install_dir>/El-MAVEN/bin/El-MAVEN.app/Contents/MacOS/
+    QDir applicationDir(QCoreApplication::applicationDirPath());
+
+#ifdef __OSX_AVAILABLE
+    applicationDir.cdUp(); // <install_dir>/El-MAVEN/bin/El-MAVEN.app/Contents/
+    applicationDir.cdUp(); // <install_dir>/El-MAVEN/bin/El-MAVEN.app/
+    applicationDir.cdUp(); // <install_dir>/El-MAVEN/bin/
+#endif
+    applicationDir.cdUp(); // <install_dir>/El-MAVEN/
+    applicationDir.cdUp(); // <install_dir>/
+
+    QString maintenanceToolPath = "";
+#ifdef __OSX_AVAILABLE
+    maintenanceToolPath = applicationDir.path()
+                          + QDir::separator()
+                          + "maintenancetool.app/Contents/MacOS/maintenancetool";
+#endif
+#ifdef Q_OS_WIN
+    maintenanceToolPath = applicationDir.path()
+                          + QDir::separator()
+                          + "maintenancetool.exe";
+#endif
+
+    qDebug() << "Using path"
+             << maintenanceToolPath
+             << "for maintenancetool executable.";
+    if (!QFile::exists(maintenanceToolPath)) {
+        qWarning() << "Maintenance tool was not found at its expected path. "
+                      "Cannot proceed with update process.";
+        return;
+    }
+
+    QProcess* maintenanceToolProcess = new QProcess;
+    connect(maintenanceToolProcess,
```

Comment:
Who is watching these two signals? `maintenanceToolStarted` and `maintenanceToolClosed`?

Reply:
Right now, nothing. But I felt like there should be some hooks (for future uses) that lets the application know that the tool was started and then closed.